src/charm.py

```diff
@@ -1338,6 +1352,23 @@ def _on_update_status_early_exit_checks(self, container) -> bool:
         if not container.can_connect():
             logger.debug("on_update_status early exit: Cannot connect to container")
             return False
+
+        self._check_pgdata_storage_size()
+
+        if self._has_blocked_status or self._has_non_restore_waiting_status:
+            # If charm was failing to disable plugin, try again and continue (user may have removed the objects)
+            if self.unit.status.message == EXTENSION_OBJECT_MESSAGE:
+                self.enable_disable_extensions()
+                return True
+
+            if (
+                self.unit.status.message == MOVE_RESTORED_CLUSTER_TO_ANOTHER_BUCKET
+                and "require-change-bucket-after-restore" not in self.app_peer_data
+            ):
```

Comment:
this extra condition is meant to allow `MOVE_RESTORED_CLUSTER_TO_ANOTHER_BUCKET` blocked status to resolve inside `update_status` event, in case the flag is no longer there, avoiding potential endless block.